src/shared/blockdev-util.c

```diff
@@ -509,19 +509,26 @@ int block_device_resize_partition(
         return RET_NERRNO(ioctl(fd, BLKPG, &ba));
 }
 
-int block_device_remove_all_partitions(int fd) {
-        struct stat stat;
-        _cleanup_(sd_device_unrefp) sd_device *dev = NULL;
+int block_device_remove_all_partitions(sd_device *dev, int fd) {
         _cleanup_(sd_device_enumerator_unrefp) sd_device_enumerator *e = NULL;
+        _cleanup_(sd_device_unrefp) sd_device *dev_unref = NULL;
         sd_device *part;
         int r, k = 0;
 
-        if (fstat(fd, &stat) < 0)
-                return -errno;
+        assert(dev || fd >= 0);
 
-        r = sd_device_new_from_devnum(&dev, 'b', stat.st_rdev);
-        if (r < 0)
-                return r;
+        if (!dev) {
+                struct stat st;
+
+                if (fstat(fd, &st) < 0)
+                        return -errno;
+
+                r = sd_device_new_from_stat_rdev(&dev_unref, &st);
+                if (r < 0)
+                        return r;
```

Comment:
maybe time for a sd_device_new_from_fd()?

Reply:
Yeah, but I'd like to do that in another later PR.